### PR TITLE
Blockly Factory: Factory Utils

### PR DESCRIPTION
**Pull request by quachtina96**
# Overview
I've pulled functions out of factory.js (BlockFactory) and into factory_utils.js (FactoryUtils), to share useful generator code between the different apps within Blockly Factory. 

## factory.js and factory_utils.js
- removed console.log calls
- removed the functions in factory.js related to downloading the single blocks block def and generator stub.
- pulled code generating functionality used both in and outside of block factory out of factory.js and into new factory_utils.js file
- factory_utils.js houses code generation functions used for the block factory, workspace factory, and exporter.

## app_controller.js, block_exporter_controller.js
-  require FactoryUtils and call functions from it.

## block_exporter_tools.js
-  require FactoryUtils and call functions from it.
-  moved getDefinedBlock out to FactoryUtils so that WFactoryGenerator may use it for generating a category from imported block definitions

## block_exporter_view.js
- call the now public workspace.cleanUp() method

## index.html
- included the new factory_utils.js

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/535)

### CODE REVIEW

**Comment by quachtina96** [comment]
@picklesrus @vicng @evd2014 Sorry this CL is so big and that it mixes refactoring with smaller changes to UI and some added functionality. I wanted whatever I had done to be shared so we can be on the same page!

**Comment by picklesrus** [comment]
Yup, this is kind of big and doing lots of things :)  I actually think the code will get checked in faster if you separate them into multiple PRs (basically 1 for each of your bullet points in the commit description).  I'm going to go through and make a separate review pass with each of those bullet points in mind to make sure things are generally moving where they should.  But there is enough code that it is really hard to keep which code is part of which change all in my head at the same time.

File: demos/blocklyfactory/app_controller.js [hunk 1]
```diff
@@ -383,7 +377,6 @@ AppController.prototype.assignFactoryClickHandlers = function() {
       });
   document.getElementById('createNewBlockButton')
     .addEventListener('click', function() {
-        BlockFactory.mainWorkspace.clear();
```
Comment (picklesrus) [inline demos/blocklyfactory/app_controller.js]: why did you remove this?

File: demos/blocklyfactory/block_exporter_tools.js [hunk 2]
```diff
@@ -221,3 +210,61 @@ BlockExporterTools.prototype.generateToolboxFromLibrary
...
+  // Define the custom blocks in order to be able to create instances of
+  // them in the exporter workspace.
+  this.addBlockDefinitions(blockXmlMap);
+
+  for (var blockType in blockXmlMap) {
+    // Get block.
+    var block = FactoryUtils.getDefinedBlock(blockType, this.hiddenWorkspace);
+
+    // Get preview block XML.
+    var blockChild = Blockly.Xml.blockToDom(block);
```
Comment (picklesrus) [inline demos/blocklyfactory/block_exporter_tools.js]: why do you call this child?  Looks to me like it is the xml of the block defined above.  Or is it the xml of the child of that block and I'm just misreading?

File: demos/blocklyfactory/block_exporter_tools.js [hunk 3]
```diff
@@ -221,3 +210,61 @@ BlockExporterTools.prototype.generateToolboxFromLibrary
...
+ *
+ * @param {!Array.<string>} blockTypesOne - types of blocks.
+ *    workspace from Workspace FactoryTwo.
+ * @param {!Array.<string>} blockTypes - array of types of blocks.
+ * TODO(quachtina96): Use as helper function in handler for  'Add All Used
+ *    Blocks' Button, to figure out which blocks used in the workspace factory
+ *    are stored in the block library. Probably more efficient if blockTypesTwo
+ *    is the block types stored in library.
+ */
+BlockExporterTools.prototype.getSharedTypes_ = function(blockTypesOne, blockTypesTwo) {
```
Comment (picklesrus) [inline demos/blocklyfactory/block_exporter_tools.js]: This isn't used yet, right?  Since this PR is already big, can you leave it out for now and either add in the one where you're going to use it, or in a separate one?

**Comment by evd2014** [comment]
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 174 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzulLnm7RhwH9jSkUw:-KOzuNUtusMV4mLTmWqq:bhvpo7w) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L174)):*
> ```JavaScript
>  * @return {!Element} Xml representation of a category.
>  */
> FactoryGenerator.prototype.generateCategoryFromBlockDefs =
> ```

We should look at if there's a way we can unify this a little better with the generateCategoryFromBlockLib function. The second half of both functions look almost exactly the same (from the for loop on).

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 180 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzvIpeAJQHd5Bb8_C5:-KOzvIpfDuTvCfzRCa4G:bbm14t2) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L180)):*
> ```JavaScript
>   categoryElement.setAttribute('name', categoryName);
> 
>   var blockDefArray = goog.string.splitLimit(blockDefs, '\n\n', 500);
> ```

We should use more robust delimiters. The delimiter should be 'Blockly.Blocks' if the format is Javascript, and '{' if the format is JSON.

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 221 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuGkN1kcq_ba-h0S5:-KOzuGkOroWnFO_lJR3r:b-w4ipg0) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L221)):*
> ```JavaScript
>  * @param {string} blockDef - A single block definition.
>  */
> FactoryUtils.prototype.getBlockTypeFromJSDef_ = function(blockDef) {
> ```

These should go in the factory utils file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*

**Comment by picklesrus** [comment]
Review status: 0 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[demos/blocklyfactory/factory.js, line 221 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzyhfkOme7s7InRLyV:-KOzyhfl0iwkoqYeCjbE:bg9tqgg) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L221)):*
> ```JavaScript
>  * Generate a file from the contents of a given text area and
>  * download that file.
>  * @deprecated
> ```

see comment below.

---

*[demos/blocklyfactory/factory.js, line 233 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuT8gcfabOCOHvSOk:-KOzuT8h3099McBrjZXf:b-9lo19h) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L233)):*
> ```JavaScript
> /**
>  * Save the workspace's xml representation to a file.
>  * @deprecated since the addition of Block Exporter.
> ```

Marking something as deprecated usually happens when a public api changes and we can't go track down all callers and fix them ourselves.  In this case, this function is local to blocklyfactory, right?  If so, rather than marking as deprecated, you should just remove it and update the callers to call the function you want instead.

---

*[demos/blocklyfactory/factory.js, line 243 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzus7-WkoSFMvxHxZM:-KOzus7-WkoSFMvxHxZN:b-3ifm1u) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L243)):*
> ```JavaScript
> /**
>  * Imports xml file for a block to the workspace.
>  * @deprecated since the addition of Block Exporter.
> ```

same comment about deprecated tab as above

---

*[demos/blocklyfactory/factory.js, line 289 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuzzQ6IYUI6h_O6EA:-KOzuzzRhVhkb266l1GJ:b-b096d8) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L289)):*
> ```JavaScript
>  */
> BlockFactory.showStarterBlock = function() {
>     BlockFactory.mainWorkspace.clear();
> ```

ah, I see you moved this here. You can disregard the other comment.

---

*[demos/blocklyfactory/factory_utils.js, line 2 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzzTNvwdK-a0DlSVLx:-KOzzTNvwdK-a0DlSVLy:bmvt3yr) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory_utils.js#L2)):*
> ```JavaScript
> 'use strict';
> ```

This file is already kind of big and I'm wondering if there are better ways to group things.  For instance some of the utils are really just simple  dom manipulation  (e.g. hide, show, make visible, etc).  There is also quite a bit of code that is really just code generation.  I wonder if it could be split up?

other thought about splitting - one set of utils dealing with json manipulation, the other with javascript...

You're way more familiar with this now though so let me know what you think!

---

*[demos/blocklyfactory/factory_utils.js, line 4 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzzQtO3Qbm0rbWVwqF:-KOzzQtO3Qbm0rbWVwqG:bkjrdnu) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory_utils.js#L4)):*
> ```JavaScript
> 
> /**
>  * Namespace for Block Factory.
> ```

update description

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*

**Comment by quachtina96** [comment]
@picklesrus, this CL will be geared at the creation of factory_utils and the refactoring of factory.js. I've reverted the other unrelated changes and am making separate CLs for them right now (at least for the ones that don't depend on this one.) Thanks for being patient with me! 

---

Review status: 0 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[demos/blocklyfactory/block_exporter_tools.js, line 238 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOvJKSjjO8KW_96aWlG-r1-238:-KP0irYflOv5ZWFJ5hFf:brm2n1q) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/block_exporter_tools.js#L238)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> why do you call this child?  Looks to me like it is the xml of the block defined above.  Or is it the xml of the child of that block and I'm just misreading?
</details>
it's the xml of the block, which will be the child of the category xml. Renamed to BlockXml.

---

*[demos/blocklyfactory/block_exporter_tools.js, line 260 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOvJKSjjO8KW_96aWlG-r1-260:-KP0ixAi5O5Y3hotkumc:b-rfp5br) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/block_exporter_tools.js#L260)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> This isn't used yet, right?  Since this PR is already big, can you leave it out for now and either add in the one where you're going to use it, or in a separate one?
</details>
yes, Done!

---

*[demos/blocklyfactory/factory.js, line 221 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzyhfkOme7s7InRLyV:-KP0jSVxtpkrtVeITEfn:bb6un1j) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L221)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> see comment below.
</details>
Done

---

*[demos/blocklyfactory/factory.js, line 233 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuT8gcfabOCOHvSOk:-KP0jQVPl-IyLflSO-JP:boq7kp) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L233)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> Marking something as deprecated usually happens when a public api changes and we can't go track down all callers and fix them ourselves.  In this case, this function is local to blocklyfactory, right?  If so, rather than marking as deprecated, you should just remove it and update the callers to call the function you want instead.
</details>
Ok! Thanks Done

---

*[demos/blocklyfactory/factory.js, line 243 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzus7-WkoSFMvxHxZM:-KP0k9QLfbA4hH97NspB:b-896fix) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L243)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> same comment about deprecated tab as above
</details>
Done.

---

*[demos/blocklyfactory/factory.js, line 289 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuzzQ6IYUI6h_O6EA:-KP0kBeq43xduyY3hpMV:b-896fix) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory.js#L289)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> ah, I see you moved this here. You can disregard the other comment.
</details>
Done.

---

*[demos/blocklyfactory/factory_utils.js, line 2 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzzTNvwdK-a0DlSVLx:-KP0lWzOrY260yCLDjSq:b-zasw7) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory_utils.js#L2)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> This file is already kind of big and I'm wondering if there are better ways to group things.  For instance some of the utils are really just simple  dom manipulation  (e.g. hide, show, make visible, etc).  There is also quite a bit of code that is really just code generation.  I wonder if it could be split up?
> 
> other thought about splitting - one set of utils dealing with json manipulation, the other with javascript...
> 
> You're way more familiar with this now though so let me know what you think!
</details>
Everything above the simple dom manipulation is related to the code generation done within block factory and in exporter. I don't want to split lines 1-658 up, but I based on your feedback, I think it would be a good idea to move hide, show, makeInvisible, and makeVisible to a new file where it will live under the AppView namespace.

---

*[demos/blocklyfactory/factory_utils.js, line 4 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzzQtO3Qbm0rbWVwqF:-KP0lxjJZLkc3KN2NjWh:b-896fix) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/factory_utils.js#L4)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> update description
</details>
Done.

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 174 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzulLnm7RhwH9jSkUw:-KP0w1i8AklFwEZB8jiK:b-rkbgy0) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L174)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> We should look at if there's a way we can unify this a little better with the generateCategoryFromBlockLib function. The second half of both functions look almost exactly the same (from the for loop on).
</details>
I have implemented this and will include this (along with the code for category generation in a separate CL)

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 180 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzvIpeAJQHd5Bb8_C5:-KP0w6rZoPJdCdsCZGS2:b-da3x60) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L180)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> We should use more robust delimiters. The delimiter should be 'Blockly.Blocks' if the format is Javascript, and '{' if the format is JSON.
</details>
I tried using those delimiters and they do not split the blockDefs string into an array of block definitions. Each element in the array is an incomplete definition when those delimiters are used. I agree taht we should use more robust delimiters though. 

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 221 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzuGkN1kcq_ba-h0S5:-KP0n7-D0WYmoX8DKzMl:birbccb) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L221)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> These should go in the factory utils file.
</details>
Thanks for catching this. I initially wanted to have these in wfactory_generatory because getblockTypeFromJSDef & defineBlockWithJson are both only called in the generateCategoryFromBlockDefs function and I would have made them private.  However, I see that we may want to use these functions in other code-generating places, so I've made them public and moved them to the factory utils file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535#-:-KP0wUUmTaBAz6MnxELr:b9k4wy5)*

**Comment by evd2014** [comment]
Review status: 0 of 9 files reviewed at latest revision, 10 unresolved discussions.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 44 \[r2\]](https://reviewable.io:443/reviews/google/blockly/535#-KPEMiiEv_BP5CcR2nmS:-KPEMiiFO2mR4fHq7N6y:b-g3g9ei) ([raw file](https://github.com/google/blockly/blob/ecc6c043ad64f78c37f682a5fb8b9273a5baba7c/demos/blocklyfactory/block_exporter_controller.js#L44)):*
> ```JavaScript
>  *
>  * @param {!BlockLibrary.Storage} blockLibStorage - Block Library Storage.
>  * @param {!FactoryGenerator} factoryGenerator - Workspace Factory
> ```

factoryGenerator is a parameter in the comments but not in the actual function. I actually started writing some code for this, so maybe we can talk about it and cover changes to the exporter in a later CL? (We'll definitely need it as a parameter here at some point)

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 180 \[r1\]](https://reviewable.io:443/reviews/google/blockly/535#-KOzvIpeAJQHd5Bb8_C5:-KPEO8pfbvPxtdXVR-gc:b55xvo5) ([raw file](https://github.com/google/blockly/blob/fcb3d3ebdc5c7b39c2b231e14520f3fd9158a9b8/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L180)):*
<details><summary><i>Previously, quachtina96 (Tina Quach) wrote…</i></summary>
> I tried using those delimiters and they do not split the blockDefs string into an array of block definitions. Each element in the array is an incomplete definition when those delimiters are used. I agree taht we should use more robust delimiters though. 
</details>
We can both work on it and try to figure out something in a later CL.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*

**Comment by quachtina96** [comment]
Review status: 0 of 17 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[demos/blocklyfactory/block_exporter_controller.js, line 44 \[r2\]](https://reviewable.io:443/reviews/google/blockly/535#-KPEMiiEv_BP5CcR2nmS:-KPER1VEurMIn26eu7p9:b-7xt2v3) ([raw file](https://github.com/google/blockly/blob/ecc6c043ad64f78c37f682a5fb8b9273a5baba7c/demos/blocklyfactory/block_exporter_controller.js#L44)):*
<details><summary><i>Previously, evd2014 (Emma Dauterman) wrote…</i></summary>
> factoryGenerator is a parameter in the comments but not in the actual function. I actually started writing some code for this, so maybe we can talk about it and cover changes to the exporter in a later CL? (We'll definitely need it as a parameter here at some point)
</details>
Yes! 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*

**Comment by picklesrus** [comment]
I think something went wrong in your merge or rebase? There are now a bunch of files in this that are probably from recently submitted changes.  The diff should only reflect yours.

---

Review status: 0 of 17 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535#-:-KPFD2JOK5-Y8YhtrfeI:b-go41ud)*

**Comment by quachtina96** [comment]
Forgot to merge after fetching upstream. I have now successfully rebased this branch on the most recent develop so the diff in r4 should be correct.

Thanks!

---

Review status: 0 of 9 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535#-:-KPFIV33mpA-tAjpZ3rL:b-ehvtws)*

**Comment by picklesrus** [comment]
Review status: 0 of 9 files reviewed at latest revision, 11 unresolved discussions.

---

*[demos/blocklyfactory/index.html, line 31 \[r4\]](https://reviewable.io:443/reviews/google/blockly/535#-KPFN07ot5Jg2O133UeX:-KPFN07ot5Jg2O133UeY:b-l4ht1e) ([raw file](https://github.com/google/blockly/blob/73f6e7fdce5d20ab20e0eb4c6d0782d25a4015f4/demos/blocklyfactory/index.html#L31)):*
> ```HTML
>       blocklyFactory = new AppController();
>       blocklyFactory.init();
>       if (blocklyFactory.blockLibraryController.hasEmptyBlockLib()) {
> ```

Did you mean to add this back? (at least I think you removed it in a CL earlier today)

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 26 \[r4\]](https://reviewable.io:443/reviews/google/blockly/535#-KPFMqGV-i7I24K1_Dln:-KPFMqGV-i7I24K1_Dlo:b-27t17x) ([raw file](https://github.com/google/blockly/blob/73f6e7fdce5d20ab20e0eb4c6d0782d25a4015f4/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L26)):*
> ```JavaScript
>  * Blockly.Xml and depends on information in the model (holds a reference).
>  * Depends on a hidden workspace created in the generator to load saved XML in
>  * order to generate toolbox XML. Depends on FactoryUtils for category xml
> ```

Did you mean to add this? I don't think it does yet... at least not yet.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*

**Comment by quachtina96** [comment]
Review status: 0 of 9 files reviewed at latest revision, 10 unresolved discussions.

---

*[demos/blocklyfactory/index.html, line 31 \[r4\]](https://reviewable.io:443/reviews/google/blockly/535#-KPFN07ot5Jg2O133UeX:-KPFQT16ofDOdBJfJIUZ:btahexp) ([raw file](https://github.com/google/blockly/blob/73f6e7fdce5d20ab20e0eb4c6d0782d25a4015f4/demos/blocklyfactory/index.html#L31)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> Did you mean to add this back? (at least I think you removed it in a CL earlier today)
</details>
Thanks for catching this--I meant to remove it but added it back so that I could remove it in a separate CL that I thought would get merged after this one. Since that one is already merged, I will remove this now.

---

*[demos/blocklyfactory/workspacefactory/wfactory_generator.js, line 26 \[r4\]](https://reviewable.io:443/reviews/google/blockly/535#-KPFMqGV-i7I24K1_Dln:-KPFR13FpgmNfwkJtOHr:b-u0e2hf) ([raw file](https://github.com/google/blockly/blob/73f6e7fdce5d20ab20e0eb4c6d0782d25a4015f4/demos/blocklyfactory/workspacefactory/wfactory_generator.js#L26)):*
<details><summary><i>Previously, picklesrus wrote…</i></summary>
> Did you mean to add this? I don't think it does yet... at least not yet.
</details>
Removed--to be included in the CL in which I add category xml generation

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/535)*
